### DISCUSSION TITLE
Question: will librespot support Spotify HiFi?

### DISCUSSION THREAD

**Discussion by willemvdg** [librespot-org/librespot#669] [Feature Requests]
Hi,

Spotify announced Spotify HiFi a few weeks back: CD quality lossless audio, see https://newsroom.spotify.com/2021-02-22/five-things-to-know-about-spotify-hifi/ .
I was wondering if librespot will support this too in the future. I am using librespot as part of the moOde™ audio player (https://moodeaudio.org).

Thanks, Willem

**Comment by michaelherger** [comment]
There likely can't be an answer given now, as Spotify hasn't even given a release date yet. And before they start the service there won't be any way to adapt librespot to it.

**Reply by giggywithit** [reply]
But, the chances of adapting it if it was to come out, would be entirely possible? or unknown?

**Comment by sashahilton00** [comment]
I did poke through a recent binary for Spotify, but there doesn't appear to be anything indicating that support has already been implemented for public clients and simply needs to be switched on, so we'll have to wait until the feature is released before we can look at it further.

**Reply by willemvdg** [reply]
Thanks for the reactions, I'll have to be patient :-)

**Comment by roderickvd** [comment]
I'm super-curious what codec they are even targeting, but have not been able to find anything on the net. If they want it to be backwards compatible with existing devices, it just has to be lossy Vorbis but perhaps at 500 kbps and/or higher sampling rates. Otherwise it will require newer hardware and software support for a different codec.

Those three options pretty much sum up the extent of changes librespot would need to make. Unless they also change endpoints or anything like that. Anyway, we'll know it when we see it!

**Reply by kingosticks** [reply]
They [have said](https://newsroom.spotify.com/2021-02-22/five-things-to-know-about-spotify-hifi/) it's going to be a

> CD-quality, lossless audio format 

It'll be whatever Tidal and friends do. Spotify are not afraid of backwards compatibility and they have broken things for old hardware before. I don't believe this will be any different and it's not even that bad - it will only mean you can't use the best quality mode on your old "smart"-speaker. But if it means you then go and buy a new smart-speaker just for this feature, that's great news for Spotify's hardware partners and presumably Spotify also.

**Comment by roderickvd** [comment]
As we're gearing up to support FLAC in `new-api`, Spotify has postponed the release of HiFi for an unknown amount of time 😢 : https://community.spotify.com/t5/Live-Ideas/All-Platforms-Music-HiFi-Quality-Lossless-Streaming-16bit-44/idc-p/5324421#M238504

**Reply by michaelherger** [reply]
Why am I not surprised?...

**Comment by giggywithit** [comment]
Hype

https://www.reddit.com/r/truespotify/comments/1ch6pzv/v1236_new_hifi_leak/#lightbox

**Comment by kingosticks** [comment]
Latest in the saga https://www.androidauthority.com/spotify-deluxe-plan-3463639/

**Comment by SuisChan** [comment]
Interesting observation, it looks like Spotify is now returning heads-fa and encrypted blobs for flac(?) since when?

**Comment by sashahilton00** [comment]
Frankly the whole Hi-Fi thing is likely dead in the water. Spotify has messed everyone around for too long, their Deluxe pricing plan is not based in reality, and their UX has been falling off a cliff in recent years.

If an open source, interoperable connect implementation was created and adopted by the other streaming platforms Spotify would take a beating.

**Reply by roderickvd** [reply]
Must be patented or something, that Spotify Connect thing, and by now be so ubiquitous that hardware manufacturers just don't see the point.

Anyway, if Spotify HiFi ever appears with AES keys still served over Mercury, 0.5 (dev) is rigged to play it.
